### DISCUSSION TITLE
Error: No supported probe was found

### DISCUSSION THREAD

**Discussion by pitoszud** [japaric/discovery#439] [Q&A] [answered]
I'm getting "No supported probe was found" error on Micorobit V2. I run the following commands:
Also, after connecting the Microbit, I've got 529 error on LEDs

```
PS E:\Rust\sample_code\discovery\microbit\src\05-led-roulette> rustup target add thumbv7em-none-eabihf
info: component 'rust-std' for target 'thumbv7em-none-eabihf' is up to date
PS E:\Rust\sample_code\discovery\microbit\src\05-led-roulette> cargo build --features v2 --target thumbv7em-none-eabihf
    Finished dev [unoptimized + debuginfo] target(s) in 0.37s
PS E:\Rust\sample_code\discovery\microbit\src\05-led-roulette> cargo readobj --features v2 --target thumbv7em-none-eabihf --bin led-roulette -- --file-headers
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s
ELF Header:
  Magic:   7f 45 4c 46 01 01 01 00 00 00 00 00 00 00 00 00
  Class:                             ELF32
  Data:                              2's complement, little endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Type:                              EXEC (Executable file)
  Machine:                           ARM
  Version:                           0x1
  Entry point address:               0x101
  Start of program headers:          52 (bytes into file)
  Start of section headers:          958740 (bytes into file)
  Flags:                             0x5000400
  Size of this header:               52 (bytes)
  Size of program headers:           32 (bytes)
  Number of program headers:         4
  Size of section headers:           40 (bytes)
  Number of section headers:         24
  Section header string table index: 22
PS E:\Rust\sample_code\discovery\microbit\src\05-led-roulette> cargo embed --features v2 --target thumbv7em-none-eabihf
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
      Config default
      Target E:\Rust\sample_code\discovery\microbit\target\thumbv7em-none-eabihf\debug\led-roulette
       Error No supported probe was found
```

It looks like this error is thrown in cargo-embeded/main.rs (section below)

```
// Only automatically select a probe if there is only
                // a single probe detected.
                let list = Probe::list_all();
                if list.len() > 1 {
                    return Err(anyhow!("The following devices were found:\n \
                                    {} \
                                        \
                                    Use '--probe VID:PID'\n \
                                                            \
                                    You can also set the [default.probe] config attribute \
                                    (in your Embed.toml) to select which probe to use. \
                                    For usage examples see https://github.com/probe-rs/cargo-embed/blob/master/src/config/default.toml .",
                                    list.iter().enumerate().map(|(num, link)| format!("[{}]: {:?}\n", num, link)).collect::<String>()));
                }
                Probe::open(
                    list.first()
                        .map(|info| {
                            METADATA.lock().unwrap().probe = Some(format!("{:?}", info.probe_type));
                            info
                        })
                        .ok_or_else(|| anyhow!("No supported probe was found"))?,
                )?
```

**Comment by adamgreig** [comment]
The error means that cargo-embed can't see your micro:bit over USB. There are a couple of possible causes...

* Maybe the USB connection is bad, or the cable is power-only? Are you able to use the normal micro:bit tools to program it?
* Maybe it's a driver issue; on Windows sometimes installing Zadig drivers (https://zadig.akeo.ie/) for that device helps (though shouldn't be necessary...)

The 529 error means at some point a bad firmware was programmed onto the device; it shouldn't cause problems programming a new one, and you should be able to clear it by programming a valid firmware either from Rust or from the normal micro:bit software.

**Reply by pitoszud** [reply]
I used the cable on Rasppery Pico, so I think the cable is fine. I tried flashing V2.00 firmware, but I'm getting "error: In application programming aborted due to an out-of-bounds address." in the FAIL.TXT file.
I also tried V2.20, but nothing happened. I also tried flashing some beta versions 0256 from ARM repo, but again I'm getting the same error.
Sadly, I did cargo build/embed without checking whether the board can be programmed with makecode.

Board details
```
# DAPLink Firmware - see https://daplink.io
Build ID: alpha9-189-g5dd23001 (gcc)
Unique ID: 00000000000528339d832e6a4b56fb17000000006e052820
HIC ID: 6e052820
Auto Reset: 0
Automation allowed: 1
Overflow detection: 0
Incompatible image detection: 0
Page erasing: 0
Daplink Mode: Bootloader
Bootloader Version: 0256
Git SHA: 5dd23001a7a3199d74870790049d6686e183316c
Local Mods: 0
USB Interfaces: MSD
Bootloader CRC: 0xa60a7780
Interface CRC: 0x1acd9169
Remount count: 0
URL: https://microbit.org/device/?id=0000&v=0256&bl=1
```

**Reply by adamgreig** [reply]
I'm a bit lost to what you're trying to flash where. Which V2/V2.2/0256 firmware are you talking about? I think you shouldn't need to reprogram the programmer chip (which is the daplink firmware), only program new firmware for the main nrf52 microcontroller. Perhaps someone more experienced with the micro:bit setup will understand better what might be going wrong.

I think Makecode should still work as normal - nothing cargo-embed does will change how Makecode works, so if makecode isn't working either then something else is wrong and needs fixing outside of the Rust parts.

**Comment by pitoszud** [comment, marked as answer]
Thanks, Adam, I contacted Microbit, and they suggested flashing the newest V2.2 version from their website (0256). Apparently, some distribution partners got the boards out before they updated the firmware. I can confirm that after flashing V2.2 I don't get the error anymore.

**Reply by adamgreig** [reply]
Great, glad to hear it!
